operations/kubernetes/csm-service-upgrade-FAILED.md

```diff
@@ -0,0 +1,193 @@
+
+Traceback (most recent call last):
+  File "/srv/unbound/manager.py", line 573, in <module>
+    main()
+  File "/srv/unbound/manager.py", line 167, in main
+    resp = kea_api('POST', '/', headers=kea_headers, json=kea_request)
+  File "/srv/unbound/manager.py", line 83, in __call__
+    response = http.request(method=method, url=url, headers=headers, **kwargs)
+  File "/usr/lib/python3.8/site-packages/requests/sessions.py", line 542, in request
+    resp = self.send(prep, **send_kwargs)
+  File "/usr/lib/python3.8/site-packages/requests/sessions.py", line 655, in send
+    r = adapter.send(request, **kwargs)
+  File "/usr/lib/python3.8/site-packages/requests/adapters.py", line 507, in send
+    raise RetryError(e, request=request)
+requests.exceptions.RetryError: HTTPConnectionPool(host='cray-dhcp-kea-api', port=8000): Max retries exceeded with url: / (Caused by ResponseError('too many 503 error responses'))
+```
+Looking for the Kea we noticed that is also in bad state, it is reporting "failed to select a subnet...." messages meaning it hasn't loaded the config.
+
+```sh
+ncn-m001:~ # kubectl -n services logs cray-dhcp-kea-84b4dd8c5d-ng4mc -c cray-dhcp-kea | head
+2022-04-19 13:29:25.948 ERROR [kea-dhcp4.bad-packets/14.139702139406208] DHCP4_PACKET_NAK_0001 [hwtype=1 00:40:a6:83:51:a6], cid=[ff:a6:83:51:a6:00:01:00:01:29:ea:f0:a5:00:40:a6:83:51:a6], tid=0x4cef7c4: failed to select a subnet for incoming packet, src 10.2.0.5, type DHCPDISCOVER
+2022-04-19 13:29:25.999 ERROR [kea-dhcp4.bad-packets/14.139702139406208] DHCP4_PACKET_NAK_0001 [hwtype=1 00:40:a6:83:51:48], cid=[ff:a6:83:51:48:00:01:00:01:29:ea:f0:76:00:40:a6:83:51:48], tid=0x4ec73527: failed to select a subnet for incoming packet, src 10.2.0.5, type DHCPDISCOVER
+2022-04-19 13:29:26.023 ERROR [kea-dhcp4.bad-packets/14.139702139406208] DHCP4_PACKET_NAK_0001 [hwtype=1 00:40:a6:83:02:8b], cid=[ff:a6:83:02:8b:00:01:00:01:29:ea:f0:65:00:40:a6:83:02:8b], tid=0x18dba25f: failed to select a subnet for incoming packet, src 10.2.0.5, type DHCPDISCOVER
+```
+
+
+Kea is getting a 503 when trying to communicate with SLS so it can't retrieve the list of networks to build the configuration
+```sh
+/ $ /srv/kea/dhcp-helper.py
+ERROR: 503 Server Error: Service Unavailable for url: http://cray-sls/v1/search/hardware?type=comptype_cabinet
+```
+
+Checking SLS and its running fine.
+
+```sh
+ncn-m001:~ # cray sls dumpstate list --format json | head
+{
+  "Hardware": {
+    "d0w1": {
+      "Parent": "d0",
+      "Xname": "d0w1",
+      "Type": "comptype_cdu_mgmt_switch",
+      "Class": "Mountain",
+      "TypeString": "CDUMgmtSwitch",
+      "LastUpdated": 1618431190,
+      "LastUpdatedTime": "2021-04-14 20:13:10.082771 +0000 +0000",
+      
+```
+It can't be queried from the Kea pod
+
+```sh
+$ curl http://cray-sls/v1/search/hardware?type=comptype_cabinet
+upstream connect error or disconnect/reset before headers. reset reason: connection failure
+```
+
+The same query from Unbound works OK
```

Comment:
6. Check if SLS can be queried from Unbound.
    
    In this example, the query worked.

    ```sh
    $ curl http://cray-sls/v1/search/hardware?type=comptype_cabinet
    [{"Parent":"s0","Children":["x3000c0r42b0","x3000c0r41b0","x3000c0r39b0","x3000m0","x3000c0r40b0"] 
    ```